feat: support multiple versions of the pg-graphql extension

jfroche opened the pull request:
Build multiple versions of the pg_graphql extension on different PostgreSQL versions.

Add test for the extensions and their upgrade on PostgreSQL 15 and 17.

Make sure we build cargo extension and cargo-pgrx with the specified Rust version.

jfroche committed:
feat(pg_graphql): build multiple versions of the pg_graphql extension

Build multiple versions of the pg_graphql extension on different PostgreSQL versions.
Add test for the extensions and their upgrade on PostgreSQL 15 and 17.

jfroche committed:
feat(cargo-pgrx): build extensions with specified Rust version

This change allows developers to target specific Rust versions for
building extensions. It implements support for building cargo extensions
and `cargo-pgrx` using the specified Rust version.

jfroche committed:
Build pg_graphql 1.4.2 using pgrx 0.10.2

jfroche committed:
Build pg_graphql 1.4.1 using pgrx 0.10.2

jfroche committed:
Build pg_graphql 1.2.3 using pgrx 0.9.7

jfroche committed:
Build pg_graphql 1.2.2 using pgrx 0.9.5

jfroche committed:
Build pg_graphql 1.2.0 using pgx 0.7.1

jfroche committed:
Build pg_graphql 1.1.0 using pgx 0.6.1

jfroche committed:
Refactoring and build pg_graphql 1.0.2 using pgx 0.6.1

jfroche committed:
chore: add release suffix for testing

jchancojr reviewed (approved):
lgtm

samrose reviewed (changes requested):
Need to finish rebase and testing

samrose committed:
fix: conditional for using this linking only with macos

samrose committed:
fix: bindgen

jfroche committed:
fix: apply bindgen fix for aarch64 to versions up to 1.5.1

The issue we face when building pg_graphql on aarch64 is:

`Invalid or unknown abi 16 for function "_ZGVnN4vv_atan2f"`

It has been fixed in bindgen 0.11.3.

jfroche committed:
feat: run pg_regress tests during build

We don't recompile the extension using pgx with dev/debug symbols, but
we do run the pg_regress tests to ensure everything is working
correctly.

jfroche committed:
chore: use default nixos test to test pg_graphql extension

jfroche committed:
fix: do not apply bindgen fix for version 1.5.4

pg_graphql 1.5.4 uses pgrx 0.11.3 which fixed the ABI issues on aarch64-linux.

jfroche committed:
fix(pg_graphql): migration script from 1.5.1-mergeless to 1.5.4

Add special case handling for 1.5.1-mergeless to 1.5.4 upgrade path.
Also replace CREATE FUNCTION with CREATE OR REPLACE FUNCTION to prevent conflicts
in migration scripts and the same for event triggers.

samrose committed:
chore: bump to release

samrose reviewed (approved):
passing all tests
